subsys/bluetooth/host/iso.c

```diff
@@ -757,13 +757,15 @@ int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf)
 		return -ENOTCONN;
 	}
 
+	iso_conn = chan->iso;
+
 	hdr = net_buf_push(buf, sizeof(*hdr));
-	hdr->sn = sys_cpu_to_le16(sn++);
+	hdr->sn = sys_cpu_to_le16(iso_conn->iso.seq_num++);
```

Comment:
It's a bad idea to include operations in the endianness macro parameter. For big endian architectures, the expression is executed more than once, and the consequence in this case is double increments and a random sn.

Reply:
Thanks for catching that! I had completely forgotten that `sys_cpu_to_le16` is a macro and not a function :)